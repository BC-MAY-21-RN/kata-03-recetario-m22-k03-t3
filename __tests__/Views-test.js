import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Home from '../src/views/home/home';
import Detail from '../src/views/detail/detail';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('Views', () => {
  it('Home', () => {
    const renderHome = renderer.create(<Home />).toJSON();
    expect(renderHome).toMatchSnapshot();
  });
  it('Detail', () => {
    const renderDetail = renderer
      .create(
        <Detail
          route={{params: {mode: true, dish: {servings: 1, ingredients: []}}}}
          navigation={{goBack: () => {}}}
        />,
      )
      .toJSON();
    expect(renderDetail).toMatchSnapshot();
  });
});
