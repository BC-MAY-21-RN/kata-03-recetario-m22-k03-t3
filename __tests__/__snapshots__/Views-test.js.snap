// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views Detail 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#282828",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <View>
    <View>
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": 385,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(43,41,41,0.53)",
            "display": "flex",
            "flexDirection": "column",
            "height": 385,
            "justifyContent": "flex-end",
            "paddingBottom": 42.5,
            "paddingHorizontal": 26.8,
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "height": 64,
              "left": 25,
              "position": "absolute",
              "right": 28,
              "top": null,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            < />
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginRight": 15,
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            < />
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            < />
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "lineHeight": 22,
              "marginBottom": 2.4,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 25,
              "lineHeight": 30,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "padding": 20.3,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
            "lineHeight": 24,
            "marginBottom": 6.5,
          }
        }
      >
        Ingredients
      </Text>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "lineHeight": 22,
          }
        }
      >
        for 
        1
         servings
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Views Home 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#282828",
      "height": "100%",
      "paddingHorizontal": 16,
      "paddingTop": 72,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(142, 142, 147, 0.12)",
          "borderRadius": 10,
          "display": "flex",
          "flexDirection": "row",
          "height": 36,
          "paddingLeft": 8,
          "paddingVertical": 0,
          "width": "100%",
        }
      }
    >
      < />
      <TextInput
        placeholder="What do you want to eat?"
        placeholderTextColor="#ffffff"
        style={
          Object {
            "color": "#fff",
            "flex": 1,
            "fontSize": 17,
            "height": "100%",
            "lineHeight": 20,
            "paddingLeft": 8,
            "paddingVertical": 0,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        < />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": 36,
          "marginBottom": 18,
          "marginTop": 22,
          "opacity": 1,
          "width": 142,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4422C5",
            "borderRadius": 40,
            "display": "flex",
            "height": 36,
            "justifyContent": "center",
            "width": 36,
          }
        }
      >
        < />
      </View>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 15,
            "lineHeight": 18,
            "marginLeft": 14,
          }
        }
      >
        Normal Mode
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#f91a81",
            "fontSize": 21,
            "marginBottom": 18.48,
          }
        }
      >
        TRENDING
      </Text>
      <RCTScrollView
        horizontal={true}
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 160,
                "marginRight": 21.5,
                "opacity": 1,
                "width": 110,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://img.bestrecipes.com.au/f4jy177E/w643-h428-cfill-q90/br/2020/11/beef-stew-960642-2.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 10,
                  "height": 110,
                  "width": 110,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                  "marginTop": 13.8,
                }
              }
            >
              Beef stew
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 160,
                "marginRight": 21.5,
                "opacity": 1,
                "width": 110,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://w6h5a5r4.rocketcdn.me/wp-content/uploads/2019/05/espaguetis-a-la-bolonesa-1080x671.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 10,
                  "height": 110,
                  "width": 110,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                  "marginTop": 13.8,
                }
              }
            >
              Spaghetti Bolognese
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 160,
                "marginRight": 21.5,
                "opacity": 1,
                "width": 110,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://images.aws.nestle.recipes/resized/57bb23531c3f01075d66db6daee4e535_CREMA_DE_BROCOLI_1200_600.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 10,
                  "height": 110,
                  "width": 110,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                  "marginTop": 13.8,
                }
              }
            >
              Cream of broccoli
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 160,
                "opacity": 1,
                "width": 110,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://www.elespectador.com/resizer/yIEnsMpz3-O4q0bIPVeu3e7zCeM=/920x613/filters:format(jpeg)/cloudfront-us-east-1.images.arcpublishing.com/elespectador/ORYS2WQ4XZHSHLVJ7E7PUZSOV4.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 10,
                  "height": 110,
                  "width": 110,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                  "marginTop": 13.8,
                }
              }
            >
              Pancakes
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "height": 25,
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "color": "#f91a81",
            "fontSize": 21,
            "marginBottom": 18.48,
          }
        }
      >
        RECENT
      </Text>
      <RCTScrollView
        horizontal={true}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;
